src/contrib/persistence/Akka.Persistence.Sql.Common/Journal/ITimestampProvider.cs

```diff
@@ -34,4 +35,16 @@ public sealed class DefaultTimestampProvider : ITimestampProvider
         /// <returns>TBD</returns>
         public long GenerateTimestamp(IPersistentRepresentation message) => DateTime.UtcNow.Ticks;
     }
+
+    public static class TimestampProviderProvider 
+    {
+        public static ITimestampProvider GetTimestampProvider(string typeName, IActorContext context)
+        {
+            var type = Type.GetType(typeName, true);
+            var withSystem = type.GetConstructor(new[] { context.System.GetType() }) != null;
+            return withSystem ?
+                (ITimestampProvider)Activator.CreateInstance(type, context.System) :
+                (ITimestampProvider)Activator.CreateInstance(type);
+        }
+    }
```

Comment:
Unit tests are currently failing:

```
  ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
  ---> System.ArgumentNullException: Value cannot be null. (Parameter 'typeName')
    at System.RuntimeType.GetType(String typeName, Boolean throwOnError, Boolean ignoreCase, StackCrawlMark& stackMark)
    at System.Type.GetType(String typeName, Boolean throwOnError)
    at Akka.Persistence.Sql.Common.Journal.TimestampProviderProvider.GetTimestampProvider(String typeName, IActorContext context) in D:\a\1\s\src\contrib\persistence\Akka.Persistence.Sql.Common\Journal\ITimestampProvider.cs:line 45
```

I think you need to return a default provider here that works the way things do now, in the case the string is null or empty.

Reply:
Man, I totally missed that. Good catch.

Reply:
Theoretically the `timestamp-provider` config value always has the default value of `Akka.Persistence.Sql.Common.Journal.DefaultTimestampProvider`. So there should always be a timestamp provider. 

But still I think it would indeed better to check `type` here as well and return an instance of Akka.Persistence.Sql.Common.Journal.DefaultTimestampProvider if null or empty.